profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java

```diff
@@ -251,7 +252,26 @@ public Object getConnectionStatus(EntityReference ref, Map<String, Object> param
 		
 		return connectionsLogic.getConnectionStatus(uuid, parameters.get("friendId").toString());
 	}
-	
+
+    @EntityCustomAction(action="unreadMessagesCount",viewKey=EntityView.VIEW_SHOW)
+	public Object getUnreadMessagesCount(EntityReference ref) {
+
+		if(!sakaiProxy.isLoggedIn()) {
+			throw new SecurityException("You must be logged in to get the unread messages count.");
+		}
+
+		//convert input to uuid
+		String uuid = sakaiProxy.ensureUuid(ref.getId());
+		if(StringUtils.isBlank(uuid)) {
+			throw new EntityNotFoundException("Invalid user.", ref.getId());
+		}
+        
+        if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
+			throw new SecurityException("You can only view your own message count.");
+        }
+
+		return new ActionReturn(messagingLogic.getAllUnreadMessagesCount(uuid));
+	}
```

Comment:
Bloody hard tabs. I'll rejig it.

On 8 March 2016 at 10:18, Steve Swinsburg <notifications@github.com> wrote:

> In
> profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java
> <https://github.com/sakaiproject/sakai/pull/1883#discussion_r55339888>:
>
> > +		if(!sakaiProxy.isLoggedIn()) {
> > +			throw new SecurityException("You must be logged in to get the unread messages count.");
> > +		}
> > +
> > +		//convert input to uuid
> > +		String uuid = sakaiProxy.ensureUuid(ref.getId());
> > +		if(StringUtils.isBlank(uuid)) {
> > +			throw new EntityNotFoundException("Invalid user.", ref.getId());
> > +		}
> > +
> > +        if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
> > +			throw new SecurityException("You can only view your own message count.");
> > +        }
> > +
> > +		return new ActionReturn(messagingLogic.getAllUnreadMessagesCount(uuid));
> > +	}
>
> Looks good. Weird spacing though bro.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/1883/files#r55339888>.
>